src/GraphQL/Utilities/Visitors/ISchemaNodeVisitor.cs

```diff
@@ -39,22 +39,22 @@ public interface ISchemaNodeVisitor
         /// <summary>
         /// Visits field of registered within the schema output graph type.
         /// </summary>
-        void VisitFieldDefinition(FieldType field, ISchema schema);
+        void VisitFieldDefinition(FieldType field, IObjectGraphType type, ISchema schema);
 
         /// <summary>
         /// Visits field of registered within the schema input graph type.
         /// </summary>
-        void VisitInputFieldDefinition(FieldType field, ISchema schema);
+        void VisitInputFieldDefinition(FieldType field, IInputObjectGraphType type, ISchema schema);
 
         /// <summary>
         /// Visits field argument of registered within the schema graph type.
         /// </summary>
-        void VisitFieldArgumentDefinition(QueryArgument argument, ISchema schema);
+        void VisitFieldArgumentDefinition(QueryArgument argument, FieldType field, IObjectGraphType type, ISchema schema);
 
         /// <summary>
         /// Visits directive argument.
         /// </summary>
-        void VisitDirectiveArgumentDefinition(QueryArgument argument, ISchema schema);
+        void VisitDirectiveArgumentDefinition(QueryArgument argument, DirectiveGraphType directive, ISchema schema);
```

Comment:
Regarding `DirectiveGraphType`. It's not really graph type:
```
 public class DirectiveGraphType : MetadataProvider, INamedType, IProvideDescription
```

We need to change the name of this type so as not to mislead anyone.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/issues/2329